apps/provider-proxy/src/services/HttpLoggerInterceptor.ts

```diff
@@ -0,0 +1,62 @@
+import { LoggerService } from "@akashnetwork/logging";
+import type { MiddlewareHandler } from "hono";
+import { Context, Next } from "hono";
+
+export interface HonoInterceptor {
+  intercept(options?: any): MiddlewareHandler;
+}
+
+type HttpRequestLog = {
+  httpRequest: {
+    requestMethod: string;
+    requestUrl: string;
+    status: number;
+    userAgent?: string;
+    referrer?: string;
+    protocol?: string;
+    remoteIp?: string;
+    duration: string;
+  };
+  fingerprint?: string;
+  userId?: string;
+};
+
+export class HttpLoggerInterceptor implements HonoInterceptor {
```

Comment:
perhaps it's a good idea to extract this into logging package to avoid duplicating?

Reply:
decorator can be applied in the api or provided with a custom provider. there's similar approach with https://github.com/akash-network/console/blob/main/apps/api/src/core/providers/http-sdk.provider.ts